server_addon/houdini/client/ayon_houdini/plugins/publish/collect_frames.py

```diff
@@ -16,86 +15,70 @@ class CollectFrames(plugin.HoudiniInstancePlugin):
-            output = output_parm.evalAtFrame(start_frame)
-        else:
-            self.log.warning("Using current frame: {}".format(hou.frame()))
-            output = output_parm.eval()
-
-        _, ext = lib.splitext(
-            output, allowed_multidot_extensions=[
-                ".ass.gz", ".bgeo.sc", ".bgeo.gz",
-                ".bgeo.lzma", ".bgeo.bz2"])
+        output = output_parm.evalAtFrame(start_frame)
         file_name = os.path.basename(output)
-        result = file_name
-
-        # Get the filename pattern match from the output
-        # path, so we can compute all frames that would
-        # come out from rendering the ROP node if there
-        # is a frame pattern in the name
-        pattern = r"\w+\.(\d+)" + re.escape(ext)
-        match = re.match(pattern, file_name)
-
-        if match and start_frame is not None:
-
-            # Check if frames are bigger than 1 (file collection)
-            # override the result
-            if end_frame - start_frame > 0:
-                result = self.create_file_list(
-                    match, int(start_frame), int(end_frame)
-                )
 
         # todo: `frames` currently conflicts with "explicit frames" for a
         #       for a custom frame list. So this should be refactored.
+
         instance.data.update({
-            "frames": result,
+            "frames": file_name,  # Set frames to the file name by default.
             "stagingDir": os.path.dirname(output)
         })
 
-    @staticmethod
-    def create_file_list(match, start_frame, end_frame):
-        """Collect files based on frame range and `regex.match`
+        # Skip unnecessary logic if start and end frames are equal.
+        if start_frame == end_frame:
+            return
+
+        # Create collection using frame pattern.
+        # e.g. 'pointcacheBgeoCache_AB010.1001.bgeo'
+        # will be <Collection "pointcacheBgeoCache_AB010.%d.bgeo [1001]">
+        frame_collection, _ = clique.assemble(
+            [file_name],
+            patterns=[clique.PATTERNS["frames"]],
+            minimum_items=1
+        )
 
-        Args:
-            match(re.match): match object
-            start_frame(int): start of the animation
-            end_frame(int): end of the animation
+        # Return as no frame pattern detected.
+        if not frame_collection:
+            return
 
-        Returns:
-            list
+        # It's always expected to be one collection.
+        frame_collection = frame_collection[0]
+        frame_collection.indexes.clear()
+        frame_collection.indexes.update(list(range(start_frame, (end_frame + 1))))
+        instance.data["frames"] = list(frame_collection)
 
-        """
 
-        # Get the padding length
-        frame = match.group(1)
-        padding = len(frame)
+class CollectHDAFrames(plugin.HoudiniInstancePlugin):
+    """Collect all frames which would be saved from HDA nodes"""
```

Comment:
This seems odd - why do we need this? An HDA doesn't make frames? What are we 'hacking' here?

Is this just so that we can get cleanup for the HDA file? isn't that particular file on disk the whole "workfile"'s actual definition of that HDA that we're trying to publish? By removing that, aren't we basically breaking the ability to go back to that workfile and cointinue on the HDA? (or is that always all encapsulated in the scene).

Also, HDA would be lacking from the settings products enum?

Reply:
> Is this just so that we can get cleanup for the HDA file?

yes, and also to have consistent instance data.

---

> isn't that particular file on disk the whole "workfile"'s actual definition of that HDA that we're trying to publish? By removing that, aren't we basically breaking the ability to go back to that workfile and cointinue on the HDA?

Yes, and also, Cleaning up HDA product type will break the scene because Houdini will no longer find the HDA file.

![image](https://github.com/ynput/ayon-core/assets/20871534/fc52bdde-f12e-4d28-a87c-aae4d6990f59)

I think I should revert changes related to HDA.

Reply:
should be resolved by https://github.com/ynput/ayon-core/pull/9/commits/60ec86206f51cbafdefaa447685e661fc2dc888a